core/src/avm2/object.rs

```diff
@@ -426,11 +475,15 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
                         id,
                         PropertyInfo::new(
                             multiname.local_name().unwrap(),
-                            multiname.namespace_set().to_vec(),
+                            vec![ns],
```

Comment:
if so, we better compare the namespace in the `add_property`.

For example, if `ns1 <= ns2`, it should not be treated as a new property in the shape

Reply:
But i think this could be acceptable, depends on what you think